[#6] 2.3.Hello,Cargo

je6bmq opened the pull request:
暫定的な翻訳を終了しました。
特に最後のLearn Rust……あたりの訳が不安なので意見いただけたら幸いです。

Commit:
Translation Start.

je6bmq committed:
Translation Start.

Commit:
暫定翻訳を終了しました。

Nnwww commented on src/doc/trpl/hello-cargo.md:
```diff
@@ -1,58 +1,54 @@
-program doesn’t have any dependencies, so we’ll only be using the first part of
-its functionality. Eventually, we’ll add more. Since we started off by using
-Cargo, it'll be easy to add later.
+Cargoはコードのビルド、依存関係の解決とビルドを行います。
```
> downloading the dependencies your code needs, and building those dependencies.

とあるので、ここは依存関係の解決よりも「依存する対象のダウンロードとそのビルド」と具体的に言ったほうが良い気がします。

Nnwww commented on src/doc/trpl/hello-cargo.md:
```diff
@@ -1,58 +1,54 @@
+Cargoはコードのビルド、依存関係の解決とビルドを行います。
+
+まずコードが依存関係を持たない場合、最初の機能のみ使用されます。
+やがてそのコードは依存関係を持つようになるでしょう。しかしCargoを使用するとそれが容易になります。
```
とすると、ここの

> しかしCargoを使用するとそれが容易になります。

の「それ」が「依存関係の解決」になるのかなと思いました。

Nnwww commented on src/doc/trpl/hello-cargo.md:
```diff
@@ -1,58 +1,54 @@
 
-To Cargo-ify our project, we need to do two things: Make a `Cargo.toml`
-configuration file, and put our source file in the right place. Let's
-do that part first:
```
> 次にソースファイルを適切な位置に保存する必要があります。

putなので保存よりも配置が適切かと思います。

Nnwww commented on src/doc/trpl/hello-cargo.md:
```diff
@@ -1,58 +1,54 @@
-want to make a library instead, we should use `lib.rs`.
-Custom file locations for the entry point can be specified
-with a [`[[lib]]` or `[[bin]]`][crates-custom] key in the TOML file described below.
+ここで注意すべき点は、実際に実行するソースファイルは名前を `main.rs` とし、ライブラリを使用する場合はライブラリのファイル名を `lib.rs`とする点です。もし他のディレクトリ名を使用したい場合は、[こちら(`[[lib]]` or `[[bin]]`)][crates-custom]を参照してTOMLファイルを設定してください。
```
"create"という言葉が使われていますし、実際コンパイル言語ではないため、

> Note that since we're creating an executable, we used `main.rs`. If we want to make a library instead, we should use `lib.rs`.

を、

> 実行形式を作成する場合、ソースファイル名を`main.rs`とするよう留意してください。ライブラリを作成するのであれば`lib.rs`としてください。

という感じにしたほうが良いのかなと思いました。

なお[本家RustBook](https://doc.rust-lang.org/book/hello-cargo.html)が更新されていました。
この行と続きの行の間に、

> This convention is required for Cargo to successfully compile our projects, but it can be overridden if we wish. 

が続きます。追訳お願いします。

Nnwww replied:
> Custom file locations for the entry point can be specified

の訳についてですが、エントリーポイントというのは重要な情報だと思いますので

> もし他のディレクトリ名を使用したい場合は

を、

> もしビルドのエントリーポイントとなるファイルのロケーションを変更したい場合は

といったように、訳に含むべきだと思います。

Nnwww commented on src/doc/trpl/hello-cargo.md:
```diff
@@ -1,58 +1,54 @@
-the top level for other things, like READMEs, license information, and anything
-not related to your code. Cargo helps us keep our projects nice and tidy. A
-place for everything, and everything in its place.
+Cargoはソースファイルが`src`ディレクトリ下に置かれるものと想定しています。READMEファイルやライセンスに関するファイルなどの他のファイルは最上位のディレクトリに置き、ソースファイルと無関係なものとします。
```
> other things, like READMEs, license information, and anything not related to your code.

とのことなので、「最上位ディレクトリに置くファイル」という主語に対して「ソースとは無関係な」という修飾が必要だと思います。従って、

> READMEファイルやライセンスに関するファイルなどの他のファイルは〜

を、例えば

> READMEファイルやライセンスに関するファイルのようなソースコードと無関係なファイルは〜

等のようにした方が良いと思われます。

Nnwww commented on src/doc/trpl/hello-cargo.md:
```diff
@@ -62,18 +58,15 @@ version = "0.0.1"
-> languages.
+このファイルは [TOML][toml] 形式のファイルです。では、TOMLとは何か説明しましょう。
+> TOMLは簡単な設定ファイルのためのフォーマットで、意味を容易に理解することができます。TOMLは、ハッシュテーブルに対して明確にマッピングを行う為に設計されました。
+>TOMLはデータ構造を様々な言語で容易に解釈されるようにされるべきです。
```
「解釈」でも問題無いですが、
恐らく「パーサなどを使った情報の取得を容易に行えるようにすべき」という意味なので「パース」という単語をそのまま使った方がプログラマにとって自然であると考えられます。従って、

> TOMLは多様な言語においてデータ構造をパースするのが容易であるべきです。

等とすると良いと思われます。

Nnwww commented on src/doc/trpl/hello-cargo.md:
```diff
@@ -82,8 +75,7 @@ $ ./target/debug/hello_world
 
-Bam! We build our project with `cargo build`, and run it with
-`./target/debug/hello_world`. We can do both in one step with `cargo run`:
+わぁ!`cargo build`と端末上で実行するだけでプロジェクトのビルドに成功しました。また、`./target/debug/hello_world` とすることで実行されます。このとき、`cargo run`としても実行することができます。
```
> We can do both in one step with `cargo run`

we can do both in one stepであることが訳文で曖昧であるため、`cargo run`とするだけでビルドと実行どちらも行ってくれることを含ませた方が良いと思われます。

Nnwww commented on src/doc/trpl/hello-cargo.md:
```diff
@@ -91,9 +83,7 @@ $ cargo run
-Notice that we didn’t re-build the project this time. Cargo figured out that
-we hadn’t changed the source file, and so it just ran the binary. If we had
-made a modification, we would have seen it do both:
+いずれ、改めてビルドする必要がないことに気付くでしょう。Cargoはソースファイルに変更が無ければバイナリをそのまま実行し、変更があればビルドと実行の両方を以下のように実行してくれます。
```
> Notice that we didn’t re-build the project this time.

は、2回目のビルドが実行されなかったことについて説明しているため、

> 今度はプロジェクトがビルドされなかったことに気付くでしょう。

等が適切であると思われます。

Nnwww commented on src/doc/trpl/hello-cargo.md:
````diff
@@ -118,14 +104,9 @@ name = "hello_world"
 ```
 
-This file is used by Cargo to keep track of dependencies in your application.
-Right now, we don’t have any, so it’s a bit sparse. You won't ever need
````
> Right now, we don’t have any, so it’s a bit sparse.

の訳が抜けていますので翻訳お願いします。

Nnwww commented on src/doc/trpl/hello-cargo.md:
```diff
@@ -135,20 +116,18 @@ $ cargo build
 
 ## A New Project
```
ここのタイトルは意図的に訳していない感じですかね？
それでも良いと思われますが念のため確認をお願いします

Commit:
指摘された点について翻訳の修正及び追訳を行いました。また、見出しについては日本語に統一しました。

Commit:
指摘された点について修正、追訳を行いました。
また、見出しは日本語に統一しました。

Nnwww commented on src/doc/trpl/hello-cargo.md:
```diff
@@ -186,20 +163,19 @@ fn main() {
-You have two options: Dive into a project with ‘[Learn Rust][learnrust]’, or
-start from the bottom and work your way up with ‘[Syntax and
-Semantics][syntax]’. More experienced systems programmers will probably prefer
-‘Learn Rust’, while those from dynamic backgrounds may enjoy either. Different
```
>  while those from dynamic backgrounds may enjoy either.

これ難しいですね…
'Learn Rust'の方に掛かっていそうでありますが…
(翻訳班のアクティブな人がほとんど居ないので、余裕があれば私の翻訳のレビューをして頂けると非常に嬉しいです…)

Commit:
"while those from dynamic backgrounds may enjoy either."
の部分について訳を追加しました。

thoseが使用されていることから、"Learn Rust"と"Syntax and Semantics"の両方を指しているのではないかと思い翻訳を行いました。
